### ISSUE TITLE
(events): configureAsyncInvoke not called

### ISSUE THREAD

**Issue by thomasstep** [aws/aws-cdk#19608]
### What is the problem?

Whenever a Lambda is added as a target to an `event.Rule` the `function.configureAsyncInvoke()` function needs to be called.

If select props (`retryAttempts`, `maxEventAge`, etc.) are included [during `Function` creation](https://github.com/aws/aws-cdk/blob/94f9d27e626bced5fc68a6ebbd653fea21c6e21e/packages/%40aws-cdk/aws-lambda/lib/function.ts#L821-L829), then an `AWS::Lambda::EventInvokeConfig` resource is created. If those same props are added during a `rule.addTarget()` call through [`events-targets.LambdaFunctionProps`](https://github.com/aws/aws-cdk/blob/94f9d27e626bced5fc68a6ebbd653fea21c6e21e/packages/%40aws-cdk/aws-events-targets/lib/lambda.ts#L8), then the `function.configureAsyncInvoke()` function is never called resulting in the `AWS::Lambda::EventInvokeConfig` resource not being created.

### Reproduction Steps

```typescript
import { Stack, Duration } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as events from 'aws-cdk-lib/aws-events';
import * as targets from 'aws-cdk-lib/aws-events-targets';
import * as lambda from 'aws-cdk-lib/aws-lambda';

const defaultCode = `exports.handler = async function (event, context, callback) {
  try {
    const data = {
      statusCode: 201,
    };
    return data;
  } catch (uncaughtError) {
    console.error(uncaughtError);
    throw uncaughtError;
  }
}`;

export class DemoStack extends Stack {

  constructor(scope: Construct, id: string) {
    super(scope, id);

    const maxEventAgeInMinutes = 3;
    const retryAttempts = 2;
    new lambda.Function(
      this,
      `${id}-function-with`,
      {
        code: lambda.Code.fromInline(defaultCode),
        handler: 'index.handler',
        runtime: lambda.Runtime.NODEJS_14_X,
        maxEventAge: Duration.minutes(maxEventAgeInMinutes),
        retryAttempts: retryAttempts,
      },
    );
    const lambdaFunctionWithoutEventInvokeConfig = new lambda.Function(
      this,
      `${id}-function-without`,
      {
        code: lambda.Code.fromInline(defaultCode),
        handler: 'index.handler',
        runtime: lambda.Runtime.NODEJS_14_X,
      },
    );
    const rule = new events.Rule(this, `${id}-rule`, {
      schedule: events.Schedule.rate(Duration.minutes(5)),
    });
    rule.addTarget(
      new targets.LambdaFunction(
        lambdaFunctionWithoutEventInvokeConfig,
        {
          maxEventAge: Duration.minutes(maxEventAgeInMinutes),
          retryAttempts: retryAttempts,
        },
      ),
    );
  }
}
```

### What did you expect to happen?

I expect a `AWS::Lambda::EventInvokeConfig` resource to be created whenever `rule.addTarget()` is called.

### What actually happened?

No `AWS::Lambda::EventInvokeConfig` resource is created.

### CDK CLI Version

2.17.0 (build f9cd009)

### Node.js Version

v16.13.1

### OS

MacOS 11.6.4

### Language

Typescript

### Language Version

TypeScript (Version 4.5.2)

**Comment by peterwoodworth** [comment]
I don't think this is a bug - this seems out of the scope of what the `eventsTargets.LambdaFunction` construct is built for (adding targets for your events)

The props (with the same names) that you're passing in to this construct are for configuring your event rule resource. If you want to configure lambda infrastructure, you'll have to create an [`EventInvokeConfig`](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-lambda.EventInvokeConfig.html) with a reference to your function, or simply pass in the necessary props to your function on creation if possible.

**Comment by thomasstep** [comment]
@peterwoodworth I appreciate the response. That flow wasn't clear to me from the documentation, but I can handle that. Thank you!
